### PR TITLE
Bump Astarte Device Sdk to 0.7.0 and update other deps

### LINKED ISSUE #177: Bump Astarte Device SDK to v0.7

### LINKED ISSUE #181: Bump Astarte Message Hub Proto to v0.6.0
Depends on https://github.com/astarte-platform/astarte-message-hub-proto/milestone/1

### PR DESCRIPTION
**Pull request by rgallor**
Changes:
- introduce astarte-message-hub-proto repo to handle proto files
- update astarte-device-sdk-rust version to 0.7.0
- update doc comments
- update dependencies

Close #177 
Close #181

### CODE REVIEW

**Review by harlem88 — COMMENTED** [review]
LGTM, good choice introducing `tokio::sync::notify` for waiting the configuration

**Comment by harlem88** [comment]
Resolve `cargo fmt --check` errors.

**Review by joshuachp — CHANGES_REQUESTED** [review]

File: examples/client/main.rs [hunk 1]
```diff
@@ -46,76 +55,125 @@ struct Cli {
...
+        .interface_directory("examples/client/interfaces")
+        .expect("failed to use interface directory")
+        .connect(grpc_cfg)
+        .await
+        .expect("failed to connect")
+        .build();
+
+    // define a Notify object to terminate the execution of the following tasks upon receiving a CTRL C input
+    let shutdown_notify = Arc::new(Notify::new());
+    let shutdown_notify_cpy = Arc::clone(&shutdown_notify);
```
[the remark this answers is not in the document]
Comment (rgallor) [reply examples/client/main.rs]: Done

File: examples/client/main.rs [hunk 2]
```diff
@@ -46,76 +55,125 @@ struct Cli {
...
+    match send_handle.await {
+        Ok(()) => {}
+        Err(err) if err.is_cancelled() => {}
+        Err(err) => error!("join error: {err:?}"),
+    };
+    match receive_handle.await {
+        Ok(()) => {}
+        Err(err) if err.is_cancelled() => {}
+        Err(err) => error!("join error: {err:?}"),
+    };
```
[the remark this answers is not in the document]
Comment (rgallor) [reply examples/client/main.rs]: Done

File: examples/message-hub-config.toml [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
+pairing_url = "http://api.astarte.localhost/pairing"
 grpc_socket_port = 50051
 interfaces_directory = "/usr/share/message-hub/astarte-interfaces/"
-pairing_token = "YOUR_PAIRING_TOKEN"
-# credentials_secret = "YOUR_CREDENTIAL_SECRET"
-astarte_ignore_ssl = false
+# pairing_token = "YOUR_PAIRING_TOKEN"
+credentials_secret = "/+5AI2Wb1NO98pnOX48sy7FfdIqwzP1MUTrACYnFzBc="
+astarte_ignore_ssl = true
 store_directory = "/var/lib/message-hub"
```
[the remark this answers is not in the document]
Comment (rgallor) [reply examples/message-hub-config.toml]: Done

File: Cargo.toml [hunk 4]
```diff
@@ -22,31 +22,36 @@ categories = ["network-programming"]
...
-zvariant = "=3.2.1"
+thiserror = "1.0.56"
+tokio = { version = "1.35.1", features = ["rt-multi-thread", "sync", "macros", "signal"] }
+tokio-stream = { version = "0.1.14", features = ["net"] }
+tokio-util ="0.7.10"
+toml = "0.5.9"
+tonic = "0.10.2"
+uuid = "1.7.0"
+zbus = { version = "3.14.1", default-features = false, features = ["tokio"] }
+# pinned transitive dependenciees
```
[the remark this answers is not in the document]
Comment (rgallor) [reply Cargo.toml:46]: Done

**Comment by joshuachp** [comment]
Update the Cargo.lock to fix the ahash dependency

File: examples/client/README.md [hunk 5]
```diff
@@ -55,9 +55,9 @@ With `send-data` we can publish new values on a server-owned interface of our lo
 instance.
 The syntax is the following:
 ```
-astartectl appengine --appengine-url http://localhost:4002/                       \
-    --realm-management-url http://localhost:4000/ --realm-key <REALM>_private.pem \
-    --realm-name <REALM> devices send-data <DEVICE_ID>                            \
+astartectl appengine --appengine-url http://api.astarte.localhost/appengine            \
+    --realm-management-url http://api.astarte.localhost/realmmanagement                \
+    --realm-key <REALM>_private.pem --realm-name <REALM> devices send-data <DEVICE_ID> \
```
Comment (harlem88) [inline examples/client/README.md:60]: These changes differ from the other API URLs used in this Readme, so put back the previously configured URL.

Comment (rgallor) [reply examples/client/README.md:60]: While rebasing I forgot to change also the other APIs, but for tests I used the new ones.

**Review by joshuachp — APPROVED** [review]

**Comment by joshuachp** [comment]
We should separate the `AstarteHandle` sender and publisher in two different objects, I will add it in a subsequent PR

File: examples/client/README.md (hunk 5 above)
Comment (harlem88) [reply examples/client/README.md:60]: Update also this [README.md](https://github.com/astarte-platform/astarte-message-hub/blob/master/examples/README.md)

**Review by harlem88 — CHANGES_REQUESTED** [review]

**Review by harlem88 — APPROVED** [review]

**Review by harlem88 — APPROVED** [review]

**Pull request merged**